src/components/Input/Input.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { FC, useState } from "react";
+import styled, { css } from "styled-components";
+import { Color } from "../../theme/util";
+import { Row, Box, Spacer, Column } from "../../theme/custom-styled-components";
+import { ErrorIcon, CheckIcon } from "../Icon/Icons";
+import { InputProps, StyledLabelProps, StyledInputProps, LabelContainerProps } from "./InputTypes";
+
+const StyledLabel = styled.label<StyledLabelProps>`
+  font-family: sans-serif;
+  z-index: 1;
+  padding: 16px 5px 16px 0;
+  font-size: 17px;
+  color: ${Color.TextPrimary};
+
+  ${(p) =>
+    p.hasError &&
+    css`
+      color: ${Color.ErrorRed};
+    `}
+  /* TODO remove 'all', animate only 'fontsize' and 'top'*/
+    transition: all 0.3s ease;
+`;
+
+const LabelContainer = styled.div<LabelContainerProps>`
+  display: flex;
+  flex-direction: row;
+  pointer-events: none;
+  position: absolute;
+  align-items: center;
+  z-index: 6;
+  pointer-events: ${(p) => (p.disabled ? "none" : "auto")};
+  padding-left: 16px;
+
+  ${(props) =>
+    props.hasFocus &&
+    css`
+      top: ${(p: { size: number }) => `-${p.size / 4 - 3}px`};
+      padding-left: 5px;
+      margin-left: 11px;
+      background: ${Color.BackgroundMain};
+      height: 16px;
+
+      ${StyledLabel} {
+        font-size: 14px;
+        padding-left: 0;
+        padding-top: 0;
+        padding-bottom: 0;
+      }
+
+      svg {
+        background: ${Color.BackgroundMain};
+        transform: scale(0.85);
+        transition: transform 0.3s ease;
+      }
+    `};
+`;
+
+const ErrorText = styled.h1`
+  font-family: sans-serif;
+  font-size: 17px;
+  color: ${Color.ErrorRed};
+  font-weight: normal;
+`;
+
+const StyledInput = styled.input<StyledInputProps>`
+  position: relative;
+  height: ${(p) => `${p.size}px`};
+  width: 100%;
+  padding: 16px;
+  border: 2px solid ${Color.BorderGrey};
+  border-radius: 7px;
+  background: ${Color.BackgroundMain};
+  font-size: 17px;
+  cursor: ${(p) => (p.disabled ? "default" : "pointer")};
+
+  :disabled {
+    border: 2px solid ${Color.BorderDisabled};
+    background: ${Color.BackgroundDisabled};
+  }
+`;
+
+const IconContainer = styled(Row)<{ disabled?: boolean }>`
+  ${(p) =>
+    p.disabled &&
+    css`
+      svg * {
+        fill: ${Color.DisabledGrey};
+      }
+    `}
+`;
+
+// Pass accurate input type
+// https://www.w3.org/WAI/tutorials/forms/validation/#validating-common-input
+
+export const Input: FC<InputProps> = ({
+  className,
+  disabled = false,
+  label,
+  onInputChange,
+  name,
+  tabIndex,
+  type = "text",
+  required,
+  maxLength,
+  pattern,
+  size = "large",
+  width = "300px",
```

Comment:
The width utility from styled-system will convert any numbers greater than 1 to a pixel value, so setting the default value to `300` is sufficient without the unit, then change the type-checking to accept `number | string`.